### DISCUSSION TITLE
Can a single-file definition use _module.args any more gracefully than this?

### DISCUSSION THREAD

**Discussion by shanesveller** [hercules-ci/flake-parts#83] [Q&A] [answered]
Somewhat spurred by skimming https://github.com/hercules-ci/flake-parts/pull/82 which I think will address the most common reason I'd want to do the following anyway.

Looking at this through the lens the [current default flake template](https://github.com/hercules-ci/flake-parts/blob/d591857e9d7dd9ddbfba0ea02b43b927c3c0f1fa/template/default/flake.nix#L8-L33), is this the main way to use `_module.args.pkgs` within a single-file definition?

```nix
{
  inputs = {
    nixpkgs.url = "github:NixOS/nixpkgs/nixos-unstable";
    rust-overlay.url = "github:oxalica/rust-overlay";
  };

  outputs = { self, flake-parts, nixpkgs, rust-overlay, ... }:
    flake-parts.lib.mkFlake { inherit self; } {
      imports = [
        (_: {
          perSystem = { system, ... }: {
            config._module.args.pkgs = import nixpkgs {
              inherit system;
              overlays = [ rust-overlay.overlays.default ];
            };
          };
        })
      ];
      systems = [ "x86_64-linux" "aarch64-darwin" ];
      perSystem = { pkgs, ... }: {
        # Available only from overlay
        packages.rust = pkgs.rust-bin.stable.latest.default;
      };
    };
}
```

Is this approach morally any different, or otherwise advantageous, over a normal `let`-binding at whatever set of curly-braces seems preferable?

**Comment by roberth** [comment, marked as answer]
I think your choice to define `pkgs` in a separate module illustrates the difference quite well. You wouldn't be able to access `pkgs` in other modules if it was in a let binding. (Other modules being in other files as well; defining anonymous modules in the same file is not something to encourage, partly because the file name may be lost in error messages)

Even nicer would be to implement
 - #74 

#82 is intended for overlay publishing; not overlay consumption.

**Reply by shanesveller** [reply]
Thanks, lots of advice in here to internalize. 🍻
